packages/cli/src/api/catalog.ts

```diff
@@ -404,6 +404,48 @@ export function getCatalogForFile(file: string, catalogs: Array<Catalog>) {
   return null
 }
 
+/**
+ * Create catalog for merged messages.
+ */
+export function getCatalogForMerge(config: LinguiConfig) {
+  const catalogConfig = config
+
+  if (catalogConfig.mergePath.endsWith(path.sep)) {
+    const extension = getFormat(config.format).catalogExtension
+    const correctPath = catalogConfig.mergePath.slice(0, -1)
+    const examplePath =
+      correctPath.replace(
+        LOCALE,
+        // Show example using one of configured locales (if any)
+        (config.locales || [])[0] || "en"
+      ) + extension
+    throw new Error(
+      // prettier-ignore
+      `Remove trailing slash from "${catalogConfig.mergePath}". Catalog path isn't a directory,` +
+          ` but translation file without extension. For example, catalog path "${correctPath}"` +
+          ` results in translation file "${examplePath}".`
+    )
+  }
+
+  // catalog name is the last directory of catalogPath.
+  // If the last part is {locale}, then catalog doesn't have an explicit name
+  const name = (function () {
+    const _name = catalogConfig.mergePath.split(path.sep).slice(-1)[0]
```

Comment:
Make sure to work with "/" only to avoid issues with Windows.
```suggestion
    const _name = normalizeRelativePath(catalogConfig.mergePath).split(PATHSEP).slice(-1)[0]
```

Reply:
Also what's up with that `slice(-1)[0]`? Can you add some explanation? It's terribly cryptic without understanding the code.

Reply:
`slice(-1)[0]` returns last element in a list, e.g. `[1, 2, 3].slice(-1)[0] === 3`

Reply:
Well, what about `path.basename` instead of this cryptic thing? :)